corpus/expressions.txt

```diff
@@ -910,23 +910,26 @@ empty lambda expression
 ==============
 
 lambda {}
-lambda { lambda }
 
 ---
 
 (program
-  (lambda (block))
-  (lambda (block (lambda))))
+  (lambda (block)))
 
 ==================
-lambda expression with body
+lambda expressions
 ==================
 
 lambda { foo }
+lambda(&block) { foo }
+lambda(&lambda{})
 
 ---
 
-(program (lambda (block (identifier))))
+(program
+  (lambda (block (identifier)))
+  (method_call (identifier) (argument_list (block_argument (identifier))) (block (identifier)))
```

Comment:
These aren't technically correct as `lambda(&block)` is a valid way to initialize a lambda, and not a method call, but we at least parse all the edge cases of `lambda` and `proc` keywords.